server/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java

```diff
@@ -82,6 +85,26 @@ public void writeTo(StreamOutput out) throws IOException {
             }
         }
 
+        @Override
+        public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
+            if (isFailure()) {
+                return Iterators.concat(ChunkedToXContentHelper.startObject(), Iterators.single((b, p) -> {
+                    ElasticsearchException.generateFailureXContent(b, p, Item.this.getFailure(), true);
+                    return b;
+                }),
+                    Iterators.single((b, p) -> b.field(Fields.STATUS, ExceptionsHelper.status(Item.this.getFailure()).getStatus())),
+                    ChunkedToXContentHelper.endObject()
```

Comment:
Should this endObject be outside the if-else block? Or removes? The original logic has it in the items loop.
If this is wrong, maybe we need a test to cover this.

Reply:
The startObject is inside as well, it's just that the code formatting obscures it.  If I change `ElasticsearchException.generateFailureXContent` to return its XContentBuilder then everything fits into one line and is much easier to read, so I'll do that.